File: 1460-number-of-substrings-containing-all-three-characters/number-of-substrings-containing-all-three-characters.cpp
class Solution {
public:
    int numberOfSubstrings(string s) {
        int indx_a=-1,indx_b=-1,indx_c=-1;
        int count =0;

        for(int i=0;i<s.length();i++)
        {
            if(s[i]=='a')
            indx_a = i;
            if(s[i]=='b')
            indx_b = i;
            if(s[i]=='c')
            indx_c = i;
            
            if(indx_a!=-1 && indx_b!=-1 && indx_c!=-1)
            {
                count = (1+min(min(indx_a,indx_b),indx_c)) + count;

            }
            

            
        }
        return count;

    }
};

File: 1460-number-of-substrings-containing-all-three-characters/number-of-substrings-containing-all-three-characters.py
class Solution:
    def numberOfSubstrings(self, s: str) -> int:
        ind_a,ind_b,ind_c = -1,-1,-1

        count = 0

        for i in range(len(s)):
            if s[i] == 'a':
                ind_a = i
            elif s[i] == 'b':
                ind_b = i
            elif s[i] == 'c':
                ind_c = i
            
            if ind_a!=-1 and ind_b!=-1 and ind_c!=-1:
                count += 1+ min(ind_a,ind_b,ind_c)
        return count

